### DISCUSSION TITLE
MVVM 异步跨线程有什么好办法更新UI

### DISCUSSION THREAD

**Discussion by ErickWei** [BYJRK/DotNet-Discussions#55] [WPF 开发]
你好，请问除了用Dispatcher还有其他更加优雅的方法吗？

**Comment by BYJRK** [comment]
如果你确实是在非 UI 线程上操作某些用于绑定的变量，那么基本上是没有办法的，只能使用 Dispatcher；

但如果你是有一些异步任务，那么只要这些任务默认的 `ConfigureAwait` 是 `True`，那么在 `await` 后本身就会回到 UI 线程，之后再进行操作就行了，不需要调用 Dispatcher

**Reply by ErickWei** [reply]
谢谢你的回复。
目前的问题就是有一些操作在异步任务里面。
比如，UI控件绑定到ViewModel的列表，异步从外部(数据库或者Excel)读取数据（耗时），需要实时添加数据到ViewModel列表中。

有什么方法  可以避免在异步中操作ViewModel中绑定到UI的数据吗 ？

**Reply by dameng324** [reply]
用Dispatcher就好，网上对于Dispatcher的介绍一大堆。
简单来说就是：在需要操作UI相关数据的时候调用Dispatcher的Invoke
```csharp
this.Dispatcher.Invoke(()=>
    {
         //在这里操作UI元素。
    });
```
